rspace/src/main/scala/coop/rchain/rspace/history/HistoryRepositoryImpl.scala

```diff
@@ -253,8 +253,10 @@ final case class HistoryRepositoryImpl[F[_]: Concurrent: Parallel: Log: Span, C,
 
   override def getHistoryReader(
       stateHash: Blake2b256Hash
-  ): HistoryReader[F, Blake2b256Hash, C, P, A, K] =
-    new RSpaceHistoryReaderImpl(history.reset(root = stateHash), leafStore)(
+  ): F[HistoryReader[F, Blake2b256Hash, C, P, A, K]] =
+    for {
+      targetHistory <- history.reset(root = stateHash)
+    } yield new RSpaceHistoryReaderImpl(targetHistory, leafStore)(
```

Comment:
This _for_ comprehension uses only the functor _map_ operation, so in these cases we usually don't use it and use _map_ directly.

```scala
history
  .reset(root = stateHash)
  .map(
    new RSpaceHistoryReaderImpl(_, leafStore)(
      Concurrent[F],
      serializeC,
      serializeP,
      serializeA,
      serializeK
    )
  )
```